test: Add e2e tests for privacy check when advanced assets privacy settings toggle set to off during onboarding

chloeYue opened the pull request:
## **Description**

Add e2e tests to make sure when asset privacy settings toggle set to off during onboarding, app should not make following API calls:
```
https://token.api.cx.metamask.io/tokens/
https://bridge.api.cx.metamask.io/getAllFeatureFlags
https://on-ramp-content.api.cx.metamask.io/regions/networks
https://chainid.network/chains.json
```

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27155?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3513
https://github.com/MetaMask/MetaMask-planning/issues/3564

## **Manual testing steps**
Check code readability, make sure tests pass.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

devin-ai-integration[bot] committed:
Add advanced-functionality.spec.ts for privacy settings tests

chloeYue committed:
onboardinh class page

devin-ai-integration[bot] committed:
Migrate onboarding tests to TypeScript and implement POM pattern

devin-ai-integration[bot] committed:
Remove original JavaScript onboarding spec file

devin-ai-integration[bot] committed:
Convert onboarding.spec.js to TypeScript

chloeYue committed:
correctios

chloeYue committed:
Merge branch 'pom-migrate-onboarding' of https://github.com/MetaMask/metamask-extension into pom-migrate-onboarding

chloeYue committed:
reduce flakiness

chloeYue committed:
reduce flakiness

chloeYue committed:
Merge branch 'develop' into pom-migrate-onboarding

chloeYue committed:
fix lint

chloeYue committed:
add more log message

devin-ai-integration[bot] committed:
chore: convert privacy test files to typescript and update page object model

devin-ai-integration[bot] committed:
chore: revert changes in configuration files to match master branch

devin-ai-integration[bot] committed:
chore: revert configuration files to match develop branch

chloeYue committed:
Merge branch 'develop' into chloe-migrate-onboarding

devin-ai-integration[bot] committed:
chore: remove migrated JavaScript test files

Removed test files that were migrated to TypeScript:
- test/e2e/tests/privacy/basic-functionality.spec.js
- test/e2e/tests/privacy/onboarding-privacy.spec.js

devin-ai-integration[bot] committed:
Merge branch 'chloe-migrate-onboarding' of https://github.com/MetaMask/metamask-extension into chloe-migrate-onboarding
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

chloeYue committed:
remove unnecessary funtions

chloeYue committed:
Merge branch 'chloe-migrate-onboarding' of https://github.com/MetaMask/metamask-extension into chloe-migrate-onboarding

chloeYue committed:
fix

chloeYue committed:
fix

chloeYue committed:
Merge branch 'develop' into chloe-migrate-onboarding

chloeYue committed:
refacto switch network

chloeYue committed:
reduce flakiness

chloeYue committed:
typo

chloeYue committed:
Merge branch 'develop' into chloe-migrate-onboarding

chloeYue committed:
add advanced privavy e2e

chloeYue committed:
Merge branch 'develop' into chloe-privacy-call

chloeYue committed:
fix lint

chloeYue committed:
Merge branch 'develop' into chloe-privacy-call

chloeYue committed:
merge develop

chloeYue committed:
merge develop

chloeYue committed:
fix

chloeYue committed:
update

chloeYue committed:
update

chloeYue committed:
Merge branch 'develop' into chloe-privacy-call

chloeYue committed:
Merge branch 'main' into chloe-privacy-call

chloeYue committed:
fix

chloeYue committed:
Merge branch 'main' into chloe-privacy-call

chloeYue committed:
Merge branch 'main' into chloe-privacy-call

chloeYue committed:
resolve conflict

chloeYue committed:
update

chloeYue committed:
update

hjetpoluru committed:
Merge branch 'main' into chloe-privacy-call

chloeYue committed:
Merge branch 'main' into chloe-privacy-call

seaona commented on test/e2e/tests/privacy/advanced-functionality-privacy.spec.ts:
```diff
@@ -0,0 +1,125 @@
+          json: [{ fakedata: true }],
+        };
+      }),
+  ];
```
I think we should also add the requests to `https://accounts.api.cx.metamask.io/v2/activeNetworks?`

Currently there's a bug where even if we have the basic functionality off the requests to this endpoint  are sent regardless. It would be great if we capture this here too -- to trigger this request we need to click on the Account

https://github.com/MetaMask/metamask-extension/issues/32312

chloeYue replied:
addressed in commit https://github.com/MetaMask/metamask-extension/pull/28110/commits/82dd16a836ec322af9d47fc7211a091a9940ddd2

seaona reviewed (commented):
Overall LGTM :raised_hands:  I've added one suggestion which can expand the coverage of it by adding an extra request (there's precisely a bug happening around this at the moment). If we add this, the spec will fail until the issue is fixed --- though this is a release blocker for 12.18 so we should expect a fix this week.

I'm also okay merging this, and adding the extra request later with the fix, but then we might want to track that with a ticket. What do you think?

chloeYue committed:
address comment

chloeYue committed:
Merge branch 'main' into chloe-privacy-call

chloeYue commented:
> Overall LGTM 🙌 I've added one suggestion which can expand the coverage of it by adding an extra request (there's precisely a bug happening around this at the moment). If we add this, the spec will fail until the issue is fixed --- though this is a release blocker for 12.18 so we should expect a fix this week.
> 
> I'm also okay merging this, and adding the extra request later with the fix, but then we might want to track that with a ticket. What do you think?

I have added the check in commit [82dd16a](https://github.com/MetaMask/metamask-extension/pull/28110/commits/82dd16a836ec322af9d47fc7211a091a9940ddd2) , but yeah as the bug is not fixed, it will cause the test to fail. So i commented it, after the bug is fixed, we just need to re-enable the check. Comments also added in bug ticket here to track [track](https://github.com/MetaMask/metamask-extension/issues/32312#issuecomment-2835207778).

chloeYue committed:
Update test/e2e/tests/privacy/advanced-functionality-privacy.spec.ts

Co-authored-by: Harika <153644847+hjetpoluru@users.noreply.github.com>
